### PR TITLE
Added Image Searcher Obj C Tests for More Scenarios

### PR DESCRIPTION
**Pull request by priankakariatyml**
1. Fixed bug in error propagation when creating image segmenter.
2. Added method declaration for inference with roi.
3. Added more tests.

### CODE REVIEW

**Comment by priankakariatyml** [comment]
@khanhlvg @schmidt-sebastian Raised a PR for more objc tests

File: tensorflow_lite_support/ios/test/task/vision/image_searcher/TFLImageSearcherTests.m [hunk 1]
```diff
@@ -14,35 +14,68 @@
...
+  NSString *modelPath = [self filePathWithName:modelName extension:@"tflite"];
   TFLImageSearcherOptions *imageSearcherOptions =
-      [[TFLImageSearcherOptions alloc] initWithModelPath:self.modelPath];
+      [[TFLImageSearcherOptions alloc] initWithModelPath:modelPath];
+
+  return imageSearcherOptions;
+}
+
+- (TFLImageSearcher *)defaultImageSearcherWithModelName:(NSString *)modelName
+                                       andIndexFileName:(nullable NSString *)indexFileName {
```
Comment (khanhlvg) [inline tensorflow_lite_support/ios/test/task/vision/image_searcher/TFLImageSearcherTests.m:71]: Remove "and" in method name as in Objective-C code convention.

**Review by khanhlvg — CHANGES_REQUESTED** [review]
Thanks Prianka! I have just one minor comment.

**Review by khanhlvg — APPROVED** [review]

**Comment by khanhlvg** [comment]
@schmidt-sebastian  The https://github.com/tensorflow/tflite-support/pull/891/commits/b7af86edcc1f8eb2ab37c3dc638bd430f5bb5762 commit was added after you imported the PR. Could you add it manually to your CL?

**Comment by khanhlvg** [comment]
@priankakariatyml The PR has been merged in https://github.com/tensorflow/tflite-support/commit/949e3e4652a72f46c012ef4f508137ba7fa71b63. The last two commits can't be added to this PR. Can you pull master, patch the last two commits and send a new PR?

**Pull request closed without merging**